### PR TITLE
Added Malay language.

### PR DESCRIPTION
**Pull request by jhisham**

### CODE REVIEW

File: django/conf/locale/ms/formats.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+FIRST_DAY_OF_WEEK = 1                   # Monday
+
+# The *_INPUT_FORMATS strings use the Python strftime format syntax,
+# see https://docs.python.org/library/datetime.html#strftime-strptime-behavior
+DATE_INPUT_FORMATS = [
+    '%d/%m/%Y', '%d/%m/%y',             # '25/10/2006', '25/10/06'
+    # '%b %d %Y', '%b %d, %Y',          # 'Oct 25 2006', 'Oct 25, 2006'
+    # '%d %b %Y', '%d %b, %Y',          # '25 Oct 2006', '25 Oct, 2006'
+    # '%B %d %Y', '%B %d, %Y',          # 'October 25 2006', 'October 25, 2006'
+    # '%d %B %Y', '%d %B, %Y',          # '25 October 2006', '25 October, 2006'
```
Comment (felixxm) [inline django/conf/locale/ms/formats.py:21]: We can chop commented formats.

File: django/conf/locale/__init__.py [hunk 2]
```diff
@@ -386,6 +386,12 @@
         'name': 'Marathi',
         'name_local': 'मराठी',
     },
+    'ms': {
+        'bidi': False,
+        'code': 'ms',
+        'name': 'Malay',
+        'name_local': 'Bahasa Melayu',
```
Comment (felixxm) [inline django/conf/locale/__init__.py:393]:
```suggestion
        'name_local': 'Melayu',
```

File: django/conf/locale/ms/formats.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+]
+DATETIME_INPUT_FORMATS = [
+    '%Y-%m-%d %H:%M:%S',                # '2006-10-25 14:30:59'
+    '%Y-%m-%d %H:%M:%S.%f',             # '2006-10-25 14:30:59.000200'
+    '%Y-%m-%d %H:%M',                   # '2006-10-25 14:30'
```
Comment (felixxm) [inline django/conf/locale/ms/formats.py:26]: Shouldn't we include `%Y-%m-%d` in `DATE_INPUT_FORMATS`?

**Review by felixxm — COMMENTED** [review]
@jhisham Thanks for this patch :+1: 

Please add also en entry to the `django/conf/locale/en/LC_MESSAGES/django.po`:
```
#: conf/global_settings.py:120
msgid "Malay"
msgstr ""
```
and release notes (`4.0.txt`) to the `Internationalization` section, e.g.
```
* Added support and translations for the Malay language.
```

File: django/conf/locale/ms/formats.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# The *_FORMAT strings use the Django date format syntax,
+# see https://docs.djangoproject.com/en/dev/ref/templates/builtins/#date
+DATE_FORMAT = 'j M Y'                   # '25 Oct 2006'
+TIME_FORMAT = 'P'                       # '2:30 p.m.'
+DATETIME_FORMAT = 'j M Y, P'            # '25 Oct 2006, 2:30 p.m.'
+YEAR_MONTH_FORMAT = 'F Y'               # 'October 2006'
+MONTH_DAY_FORMAT = 'j F'                # '25 October'
+SHORT_DATE_FORMAT = 'd/m/Y'             # '25/10/2006'
+SHORT_DATETIME_FORMAT = 'd/m/Y P'       # '25/10/2006 2:30 p.m.'
+FIRST_DAY_OF_WEEK = 1                   # Monday
```
Comment (felixxm) [inline django/conf/locale/ms/formats.py:12]: Can you confirm that it's Monday, not Sunday?

File: django/conf/locale/ms/formats.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    '%d/%m/%Y %H:%M:%S',                # '25/10/2006 14:30:59'
+    '%d/%m/%Y %H:%M:%S.%f',             # '25/10/2006 14:30:59.000200'
+    '%d/%m/%Y %H:%M',                   # '25/10/2006 14:30'
+    '%d/%m/%y %H:%M:%S',                # '25/10/06 14:30:59'
+    '%d/%m/%y %H:%M:%S.%f',             # '25/10/06 14:30:59.000200'
+    '%d/%m/%y %H:%M',                   # '25/10/06 14:30'
+]
+DECIMAL_SEPARATOR = '.'
+THOUSAND_SEPARATOR = ','
+NUMBER_GROUPING = 3
```
[the remark this answers is not in the document]
Comment (jhisham) [reply django/conf/locale/ms/formats.py:34]: Hi @felixxm,

Committed changes/fixes based on comments.

- Added the mentioned entry to `django/conf/locale/en/LC_MESSAGES/django.po`
- Added an `Internationalization` section and an entry to the `4.0.txt` release notes
- Added an empty `django/conf/locale/ms/__init__.py`

File: docs/releases/4.0.txt [hunk 6]
```diff
@@ -632,6 +632,11 @@ Moreover ``USE_L10N`` is deprecated as of this release. Starting with Django
 The :ttag:`{% localize %} <localize>` tag and the :tfilter:`localize`/
 :tfilter:`unlocalize` filters will still be honored by Django.
 
+Internationalization
+--------------------
+
+* Added support and translations for the Malay language.
```
Comment (felixxm) [inline docs/releases/4.0.txt:638]: This is not a backward incompatible change but a new feature, so please move it below the [`Forms`](https://docs.djangoproject.com/en/4.0/releases/4.0/#forms):
```
Internationalization
~~~~~~~~~~~~~~~~~~~~

 * Added support and translations for the Malay language. 
```

File: django/conf/locale/ms/formats.py (hunk 1 above)
Comment (jhisham) [reply django/conf/locale/ms/formats.py]: Chopped off unused formats and retained only suitable formats.

File: django/conf/locale/__init__.py (hunk 2 above)
Comment (jhisham) [reply django/conf/locale/__init__.py:393]: `name_local` should be `Bahasa Melayu`, it's the same case as `Bahasa Indonesia`.

File: django/conf/locale/ms/formats.py (hunk 3 above)
Comment (jhisham) [reply django/conf/locale/ms/formats.py:24]: I agree. I've included it in my fix.

File: django/conf/locale/__init__.py (hunk 2 above)
Comment (felixxm) [reply django/conf/locale/__init__.py:393]: Thanks for confirmation.

File: django/conf/locale/ms/formats.py (hunk 4 above)
Comment (jhisham) [reply django/conf/locale/ms/formats.py:12]: Confirmed that it's Monday (reference: [https://savvytime.com/week-number/malaysia/2021](https://savvytime.com/week-number/malaysia/2021))

File: docs/releases/4.0.txt (hunk 6 above)
Comment (jhisham) [reply docs/releases/4.0.txt]: Moved `Internationalization` below `Forms` as requested.

File: django/conf/locale/ms/formats.py (hunk 4 above)
Comment (felixxm) [reply django/conf/locale/ms/formats.py:12]: I found sources that claim that it's Sunday :thinking: :
- https://www.timeanddate.com/calendar/?year=2021&country=69
- https://en.wikipedia.org/wiki/Names_of_the_days_of_the_week#Days_numbered_from_Sunday

Comment (felixxm) [reply django/conf/locale/ms/formats.py:12]: @sdil Can you take a look? Thanks!

Comment (jhisham) [reply django/conf/locale/ms/formats.py:12]: I'm gonna have to agree with you. I checked locally-produced calendars and they begin with Sunday. 

- [https://www.tourism.gov.my/promotional/calendar-2021](https://www.tourism.gov.my/promotional/calendar-2021)
- [https://www.permohonan.my/kalendar-cuti-umum-sekolah-malaysia/](https://www.permohonan.my/kalendar-cuti-umum-sekolah-malaysia/)
- [https://www.onestopmalaysia.com/public-holidays-2021.html](https://www.onestopmalaysia.com/public-holidays-2021.html)

Let me make the change and commit.

**Comment by felixxm** [comment]
@jhisham Thanks :+1: Welcome aboard :boat:

File: django/conf/locale/ms/formats.py (hunk 4 above)
Comment (sdil) [reply django/conf/locale/ms/formats.py]: This is a bit complicated here in Malaysia. Some states declare Sunday as the start day of the week while others declare Monday as the start of the week.

You can refer to this table: https://en.wikipedia.org/wiki/Workweek_and_weekend#Around_the_world

Based on the calendars, I agree that the start day of the week is Sunday.

Comment (felixxm) [reply django/conf/locale/ms/formats.py]: @sdil Thanks for checking :+1:

Comment (sdil) [reply django/conf/locale/ms/formats.py]: My pleasure :)

**Comment by claudep** [comment]
Thanks a lot!

**Pull request merged**